### PR TITLE
added up/down arrow steping through command history issue #32

### PR DESCRIPTION
**Pull request by qiang-bi**
I just did a rough version of issue #32 and it is functional. The only thing I have not figured out is how to have focus() actually put the cursor to the end of text after you press up key to get last input (since up arrow key move the cursor to the start of text as well..)

Please review and give comments or feel free to make changes

### CODE REVIEW

File: gdbgui/static/js/gdbgui.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ window.State = (function ($, _, Awesomplete, io, moment, debug, initial_data) {
 /**
  * Constants
  */
-const ENTER_BUTTON_NUM = 13
+const ENTER_BUTTON_NUM = 13, UP_BUTTON_NUM = 38, DOWN_BUTTON_NUM = 40
```
Comment (cs01) [inline gdbgui/static/js/gdbgui.js]: put on separate lines

File: gdbgui/static/js/gdbgui.js [hunk 2]
```diff
@@ -695,6 +695,8 @@ const GdbConsoleComponent = {
         GdbCommandInput.set_input_text(previous_cmd_from_history)
         // put focus back in input so user can just hit enter
         GdbCommandInput.el.focus()
+        //reset up-down arrow cmd history index
+        localStorage.setItem("cache-cmd-index", 0);
```
Comment (cs01) [inline gdbgui/static/js/gdbgui.js]: Single quotes everywhere `'` unless double quotes are required. Also, another convention in the file is no semicolons.

**Comment by cs01** [comment]
Very cool, thank you. I do not have time to test it today, but gave a couple quick comments. I'll look at it ASAP, likely a day or two.

**Comment by qiang-bi** [comment]
Any update on this? I see that there is a conflict from the newest branch now

**Comment by cs01** [comment]
I haven't forgotten about it. I'm planning on getting to it in the next couple days since unfortunately I couldn't do it earlier. The merge conflict is minor, it won't be hard to fix.

I wanted to store an array of commands that were sent rather than selecting the classes each time the up/down arrow is pressed. If you want to give it a shot it would save me the work, it's up to you. Otherwise I will get to it. Thanks again for contributing.

**Pull request merged**